reference-tests/src/main/java/net/consensys/linea/ReferenceTestOutcomeRecorderTool.java

```diff
@@ -35,174 +31,93 @@
-  public static void incrementSuccessRate(String jsonOutputFilename) {
-    readBlockchainReferenceTestsOutput(jsonOutputFilename)
-        .thenAccept(
-            jsonString -> {
-              BlockchainReferenceTestOutcome blockchainReferenceTestOutcome =
-                  getBlockchainReferenceTestOutcome(jsonString);
-
-              BlockchainReferenceTestOutcome finalBlockchainReferenceTestOutcome =
-                  new BlockchainReferenceTestOutcome(
-                      blockchainReferenceTestOutcome.failedCounter(),
-                      blockchainReferenceTestOutcome.successCounter() + 1,
-                      blockchainReferenceTestOutcome.modulesToConstraints());
-              jsonString = jsonConverter.toJson(finalBlockchainReferenceTestOutcome);
-              writeToJsonFile(jsonString, jsonOutputFilename);
-            });
   }
 
   @Synchronized
-  private static void mapFailedConstraintsToTestsToModule(
-      List<ModuleToConstraints> modulesToConstraints,
-      Set<String> failedConstraints,
-      String testName) {
-    for (String constraint : failedConstraints) {
-      String moduleName = getModuleFromFailedConstraint(constraint);
-
-      addModuleIfAbsent(modulesToConstraints, moduleName);
-      ModuleToConstraints moduleMapping = getModule(modulesToConstraints, moduleName);
-      String cleanedConstraintName = getCleanedConstraintName(constraint);
-
-      Set<String> failedTests =
-          aggregateFailedTestsForModuleConstraintPair(
-              testName, Objects.requireNonNull(moduleMapping), cleanedConstraintName);
-      moduleMapping.constraints().put(cleanedConstraintName, failedTests);
+  public static BlockchainReferenceTestOutcome parseBlockchainReferenceTestOutcome(
+          String jsonString) {
+    if (!jsonString.isEmpty()) {
+      BlockchainReferenceTestOutcome blockchainReferenceTestOutcome =
+              jsonConverter.fromJson(jsonString, BlockchainReferenceTestOutcome.class);
+      return blockchainReferenceTestOutcome;
     }
+    throw new RuntimeException("invalid JSON");
   }
 
-  private static Set<String> aggregateFailedTestsForModuleConstraintPair(
-      String testName, ModuleToConstraints moduleMapping, String cleanedConstraintName) {
-    Set<String> failedTests =
-        new HashSet<>(
-            moduleMapping
-                .constraints()
-                .getOrDefault(cleanedConstraintName, Collections.emptySet()));
-    failedTests.add(testName);
-    return failedTests;
-  }
-
-  private static String getCleanedConstraintName(String constraint) {
-    String cleanedConstraintName = constraint;
-    if (cleanedConstraintName.contains(".")) {
-      cleanedConstraintName = cleanedConstraintName.split("\\.")[1];
+  public static Map<String, Set<String>> extractConstraints(String message) {
+    String cleaned = message.substring(message.indexOf("constraints failed:") + "constraints failed:".length()).replaceAll(("\\[[0-9]+m"), " ").replace(']', ' ').trim();
```

Comment:
Unless something else has changed we can't just parse for for "constraints failed:" we also have to check for "failing constraint" too, there are 2 different log messages.

Reply:
this constraint failed is the summary at the end of the file, i did not see failing constraint at this stage, do you have an example ?

Reply:
It was highlighted to me by @OlivierBBB because it was identifying failing constraints when non were being caught by "constraints failed:". I only have an example of an image of a log unfortunately:
![image](https://github.com/user-attachments/assets/d30eac57-4a6e-4f34-af18-9fbbd394ab3d)

Reply:
added